src/odemis/driver/tucsen.py

```diff
@@ -0,0 +1,2530 @@
+        self._nrCameras = init_args.uiCamCount
+
+        self._lock = threading.Lock()
+
+    @staticmethod
+    def _errcheck(result, func, args):
+        """
+        Analyse the return value of a call and raise an exception in case of
+        error.
+        Follows the ctypes.errcheck callback convention
+        """
+        if result >= TUCAMRET_Enum.TUCAMRET_FAILURE.value:
+            if result in tucam_error_codes:
+                raise TUCamError(result, "Call to %s failed with error code 0x%x: %s" %
+                               (str(func.__name__), result, tucam_error_codes[result]))
+            else:
+                raise TUCamError(result, "Call to %s failed with unknown error code 0x%x" %
+                               (str(func.__name__), result))
+        return result
+
+    # hardware interaction functionality
+    # gets and sets the physical parameters
+
+    def get_info(self, id: TUCAM_IDINFO) -> str:
+        """
+        Returns info on an open camera.
+
+        Parameters
+        ----------
+        id : TUCAM_IDINFO
+            The desired information identifier,
+
+        Returns
+        -------
+        str
+
+        Raises
+        ------
+        TUCamError
+            If the actual SDK function call does not return TUCAMRET_SUCCESS
+        """
+        tvinfo = TUCAM_VALUE_INFO(id.value, 0, 0, 0)
+        self.TUCAM_Dev_GetInfo(self.TUCAMOPEN.hIdxTUCam, pointer(tvinfo))
+        return ctypes.string_at(tvinfo.pText).decode('utf-8')
+
+    def get_capability_info(self, id: TUCAM_IDCAPA) -> Tuple[float, float, float, float]:
+        """
+        Returns capability  on an open camera.
+
+        Parameters
+        ----------
+        id : TUCAM_IDCAPA
+            The desired information identifier,
+
+        Returns
+        -------
+        tuple of min, max, default, step
+
+        Raises
+        ------
+        TUCamError
+            If the actual SDK function call does not return TUCAMRET_SUCCESS
+        ValueError
+            If the requested capability does not exist.
+        """
+        #returns information about the capability, meaning its minimum, maximum, default, step.
+        if (id.value >= TUCAM_IDCAPA.TUIDC_ENDCAPABILITY.value):
+            raise ValueError("No such capability")
+        capainfo = TUCAM_CAPA_ATTR()
+        capainfo.idCapa = id.value
+        self.TUCAM_Capa_GetAttr(self.TUCAMOPEN.hIdxTUCam, pointer(capainfo))
+
+        return capainfo.nValMin, capainfo.nValMax, capainfo.nValDft, capainfo.nValStep
+
+    def set_capability_value(self, cap: TUCAM_IDCAPA, val: float) -> None:
+        """
+        Sets a capability on an open camera.
+        For a list of capabilities see the SDK,  TUCAM_IDCAPA enum
+
+        Parameters
+        ----------
+        cap : TUCAM_IDCAPA
+            The desired information identifier,
+        val : number
+            The value (float or int) to set the capability to.
+
+        Raises
+        ------
+        TUCamError
```

Comment:
This exception can never be raised as any exception will only raise `ValueError`. Other functions also have this kind of exception handling.